qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/snippets/QuteParentSnippetContext.java

```diff
@@ -69,21 +73,36 @@ public QuteParentSnippetContext read(final JsonReader in) throws IOException {
 			if (nextToken == JsonToken.NULL) {
 				return null;
 			}
-
+			String parent = null;
+			boolean unique = false;
+			String tag = null;
 			in.beginObject();
 			while (in.hasNext()) {
 				String name = in.nextName();
 				switch (name) {
 				case "parent":
 					if (in.peek() == JsonToken.STRING) {
-						return new QuteParentSnippetContext(in.nextString());
+						parent = in.nextString();
+					}
+					break;
+				case "unique":
+					if (in.peek() == JsonToken.BOOLEAN) {
+						unique = in.nextBoolean();
+					}
+					break;
+				case "tag":

```

Comment:
I'm looking for a better way to retrieve `prefix` from `QuteParentSnippetContext.isMatch`, since the `tag` field is just a repetition.

Reply:
Indeed it should be better to duplicate this tag label.

I suggest that you refactor commons Snippet classes to have that:

```
public interface ISnippetContext<T> {

	void setPrefixes(List<String> prefixes);
	
	List<String> getPrefixes();
```

In Snippet class, update setContext to fill the prefixes of the snippet:

```java
	public void setContext(ISnippetContext<?> context) {
		this.context = context;
		this.context.setPrefixes(getPrefixes());
	}

```

To help implementing ISnippetContext, create the abstract class like:

```java
package com.redhat.qute.ls.commons.snippets;

import java.util.List;

public abstract class AbstractSnippetContext<T> implements ISnippetContext<T> {

	private List<String> prefixes;

	@Override
	public void setPrefixes(List<String> prefixes) {
		this.prefixes = prefixes;
	}

	@Override
	public List<String> getPrefixes() {
		return prefixes;
	}

}

```

Create a Qute abstract class like 

```
package com.redhat.qute.services.snippets;

import com.redhat.qute.ls.commons.snippets.AbstractSnippetContext;
import com.redhat.qute.services.completions.CompletionRequest;

public abstract class AbstractQuteSnippetContext extends AbstractSnippetContext<CompletionRequest>
		implements IQuteSnippetContext {

}
```

Update your Qute snippet contexnt by extening this class:

 * `public class QuteParentSnippetContext extends AbstractQuteSnippetContext {`
 * `public class QuteSnippetContext extends AbstractQuteSnippetContext {`

After that in thei sMatch method, you can access to the prefix with super.getPrefixes(). As you havejust one prefix you must just write

`String tag = super.getPrefixes().get(0);`